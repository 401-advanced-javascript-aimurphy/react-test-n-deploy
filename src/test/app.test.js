import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
// ^renders the markup

import App from '../components/app';
import Counter from '../components/app';

describe('<Counter/>', () => {

  it('starts at 0', () => {
    let counter = mount(<Counter />);
    let count = counter.find('h4');
    // button.simulate('click');
    expect(counter.find('h4').text()).toContain('0');
  });

  it('goes up on more', () => {
    let counter = mount(<Counter />);
    // let count = counter.find('h4');
    let button = counter.find('#more')
    button.simulate('click');
    expect(counter.find('h4').text()).toContain('1');
  });

  it('goes down on less', () => {
    let counter = mount(<Counter />);
    // let count = counter.find('h4');
    let button = counter.find('#less')
    button.simulate('click');
    expect(counter.find('h4').text()).toContain('-1');
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// Assert state changes properly
// Assert that state is being transferred to the DOM


// Assert DOM stability via snapshot testing.
// i.e. take a snapshot, change the markup/jsx, assert failure. Put it back to make it all good.