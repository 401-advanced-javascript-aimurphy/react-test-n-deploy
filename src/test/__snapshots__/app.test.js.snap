// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> renders correctly 1`] = `
Array [
  <header>
    <h1>
      Header
    </h1>
  </header>,
  <div>
    <button
      id="less"
      onClick={[Function]}
    >
      LESS?
    </button>
    <h4
      className="neutral"
    >
      0
    </h4>
    <button
      id="more"
      onClick={[Function]}
    >
      MORE!
    </button>
  </div>,
  <footer>
    Â© 2018 Code Fellows
  </footer>,
]
`;
